cmd/lakefs/cmd/run.go

```diff
@@ -61,6 +62,28 @@ type Shutter interface {
 	Shutdown(context.Context) error
 }
 
+func newExternalBasicAuthenticator(cfg *config.Config, service auth.Service, logger logging.Logger) (auth.Authenticator, error) {
+
+	if cfg.Auth.RemoteAuthenticator != nil && cfg.Auth.LDAP != nil {
+		return nil, fmt.Errorf("cannot configure both LDAP (Deprecated) and RemoteAuthenticator")
+	}
+
+	if cfg.Auth.RemoteAuthenticator != nil {
+		remoteAuthenticator, err := remoteauth.NewRemoteAuthenticator(cfg.Auth.RemoteAuthenticator, service, logger)
+		if err != nil {
+			return nil, fmt.Errorf("creating new remote authenticator: %w", err)
+		}
+		return remoteAuthenticator, nil
+	} else if cfg.Auth.LDAP != nil {
```

Comment:
```suggestion
	}
	if cfg.Auth.LDAP != nil {
```

We try to prefer the "early exit" stage.  It makes code easier to read, because there is less nesting and conditions are a bit easier to remember

Reply:
done